core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -1037,4 +1037,52 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
             JSON.stringify({ keyword, languageId, archived })
         );
     }
+
+    /**
+     * Get user permission for a page or contentlet
+     *
+     * @private
+     * @param {DotCMSContentlet} item
+     * @return {*}  {UserPagePermission}
+     * @memberof DotPageStore
+     */
+    private getUserPagePermissions(item: DotCMSContentlet): UserPagePermission {
+        // Logged user
+        const { loggedUser } = this.get();
+
+        // Item types
+        const isPage = item.baseType === 'HTMLPAGE';
+        const isContent = item.baseType === 'CONTENT';
+
+        // Page permissions
+        const canUserReadPage = isPage && loggedUser.canRead.htmlPages == true;
+        const canUserWritePage = isPage && loggedUser.canWrite.htmlPages == true;
+
+        // Contentlet permissions
+        const canUserReadContent = isContent && loggedUser.canRead.contentlets == true;
+        const canUserWriteContent = isContent && loggedUser.canWrite.contentlets == true;
```

Comment:
Maybe the assertion with `true` is not needed

Reply:
Lol, yeah. Just move the logic to a function. Forgot about that.

Reply:
been there, done that